### ISSUE TITLE
terraform crash creating proxmox_vm_qemu against proxmox 6.3-6

### ISSUE THREAD

**Issue by rsalmond** [Telmate/terraform-provider-proxmox#347]
```
$ terraform version
Terraform v0.15.3
on linux_amd64
+ provider registry.terraform.io/telmate/proxmox v2.6.9
```

```.hcl
terraform {
  required_providers {
    proxmox = {
      source  = "Telmate/proxmox"
      version = "2.6.9"
    }
  }
}

provider "proxmox" {
  pm_tls_insecure = true
  pm_api_url      = "https://10.0.0.111:8006/api2/json"
  pm_user         = "root@pam"
  pm_password     = "password"
}

resource "proxmox_vm_qemu" "example" {
  name        = "servymcserverface"
  desc        = "A test for using terraform and vagrant"
  target_node = "bashful"
  iso         = "local:iso/alpine-standard-3.13.5-x86_64.iso"
}
```

```
$ terraform apply -auto-approve
proxmox_vm_qemu.example: Creating...
proxmox_vm_qemu.example: Still creating... [10s elapsed]

... snip ...

proxmox_vm_qemu.example: Still creating... [10m10s elapsed]
╷
│ Error: Plugin did not respond
│ 
│   with proxmox_vm_qemu.example,
│   on main.tf line 1, in resource "proxmox_vm_qemu" "example":
│    1: resource "proxmox_vm_qemu" "example" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-proxmox_v2.6.9 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 42 [running]:
github.com/Telmate/terraform-provider-proxmox/proxmox.initConnInfo(0xc000171d00, 0xc0003eef00, 0xc0003ee690, 0xc0002bb230, 0xc0002bb498, 0xc0002bb190, 0x0, 0x0)
	github.com/Telmate/terraform-provider-proxmox/proxmox/resource_vm_qemu.go:1608 +0x129b
github.com/Telmate/terraform-provider-proxmox/proxmox.resourceVmQemuCreate(0xc000171d00, 0xb52840, 0xc0003eef00, 0x0, 0x0)
	github.com/Telmate/terraform-provider-proxmox/proxmox/resource_vm_qemu.go:916 +0x1bd9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc00031a8f0, 0xdc1b60, 0xc0006c6540, 0xc000171d00, 0xb52840, 0xc0003eef00, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/helper/schema/resource.go:268 +0x88
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00031a8f0, 0xdc1b60, 0xc0006c6540, 0xc00017d3b0, 0xc00009d040, 0xb52840, 0xc0003eef00, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/helper/schema/resource.go:386 +0x681
github.com/hashicorp/terraform-plugin-sdk/v2/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00030a220, 0xdc1b60, 0xc0006c6540, 0xc00017d180, 0xc00030a220, 0xc00030a230, 0xccaa38)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/internal/helper/plugin/grpc_provider.go:952 +0x8b2
github.com/hashicorp/terraform-plugin-sdk/v2/internal/tfplugin5._Provider_ApplyResourceChange_Handler.func1(0xdc1b60, 0xc0006c6540, 0xc5c240, 0xc00017d180, 0xc0006c6540, 0xbdf0e0, 0xc000077001, 0xc00009c4e0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/internal/tfplugin5/tfplugin5.pb.go:3312 +0x86
github.com/hashicorp/terraform-plugin-sdk/v2/plugin.Serve.func3.1(0xdc1c20, 0xc0004005d0, 0xc5c240, 0xc00017d180, 0xc00009c4c0, 0xc00009c4e0, 0xc000356ba0, 0x5863c8, 0xc344e0, 0xc0004005d0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/plugin/serve.go:76 +0x87
github.com/hashicorp/terraform-plugin-sdk/v2/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0xc6a4c0, 0xc00030a220, 0xdc1c20, 0xc0004005d0, 0xc000077020, 0xc00045d6a0, 0xdc1c20, 0xc0004005d0, 0xc0001c5100, 0x6c4)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.1.0/internal/tfplugin5/tfplugin5.pb.go:3314 +0x14b
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00033e540, 0xdca740, 0xc000001c80, 0xc00017e200, 0xc000103da0, 0x12f0820, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.30.0/server.go:1171 +0x50a
google.golang.org/grpc.(*Server).handleStream(0xc00033e540, 0xdca740, 0xc000001c80, 0xc00017e200, 0x0)
	google.golang.org/grpc@v1.30.0/server.go:1494 +0xccd
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00030c180, 0xc00033e540, 0xdca740, 0xc000001c80, 0xc00017e200)
	google.golang.org/grpc@v1.30.0/server.go:834 +0xa1
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.30.0/server.go:832 +0x204

Error: The terraform-provider-proxmox_v2.6.9 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by tagur87** [comment]
What terraform version are you running? 

I just tested this on v0.15.1 and while it did take a long time, it didn't crash.  It seems that it's timing out while waiting for the device info to populate from the qemu-guest-agent.

You could try running terraform in trace log mode as well, to see if this gives you more indications of the reasons for the crash. `TF_LOG=TRACE terraform apply`

Another thing to try is to follow the instructions below for a cloud-init image with the guest-agent installed.

https://registry.terraform.io/modules/sdhibit/cloud-init-vm/proxmox/latest/examples/ubuntu_single_vm

**Comment by rsalmond** [comment]
I'm on terraform 0.15.3 and version v2.6.9 of the proxmox provider. I did try running with debug logging but all that revealed was [what these folks noted](https://github.com/Telmate/terraform-provider-proxmox/issues/325#issuecomment-815510175), the timeout seems to be due to waiting for the qemu guest agent to become ready. 

The stack trace suggests the fault is occurring on [this line](https://github.com/Telmate/terraform-provider-proxmox/blob/master/proxmox/resource_vm_qemu.go#L1608) which I was able to bypass by setting [define_connection_info to false](https://github.com/Telmate/terraform-provider-proxmox/blob/master/proxmox/resource_vm_qemu.go#L1571). 

My guess is that we need a sanity check in the `initConnInfo` function which verifies the `net0` field is present in the `vmConfig` map before trying to access it.

I agree that it's likely the whole issue could be sidestepped by using a guest VM with the guest agent in it but crashing when it is not present seems suboptimal.
